### DISCUSSION TITLE
lazy programmer .. want to reduce the typing in a template

### DISCUSSION THREAD

**Discussion by jmls** [hairyhenderson/gomplate#1883] [Q&A]
so, if I use a json file for input, the examples at https://docs.gomplate.ca/datasources/ show that I can use this format

![image](https://github.com/hairyhenderson/gomplate/assets/929523/eabe1948-bf25-4b9e-a888-a109be7cbd27)

However, if I have a template where the "name" is referred to in a dozen places, typing out 

{{ $person.name }} 

rather than

{{ (datasource "person").name }}

is quite appealing ;) 

Is it possible to define a var at the top of the template that stores the (datasource "person") and which can then be used throughout the template ? 

Sorry, don't know go templating at all, frantically trying to teach myself it ...

**Comment by hairyhenderson** [comment]
Apologies for the delayed response...

You can indeed [define variables](https://docs.gomplate.ca/syntax/#variables):

```
{{ $person := datasource "person" }}
Hello, {{ $person.name }}
```

Another approach is to use a [context](https://docs.gomplate.ca/syntax/#the-context) instead of a datasource, which removes the need to use the `datasource` function:

```console
$ gomplate --context person=person.json -i 'Hello {{ .person.name }}'
...
```
